scenario/state.py

```diff
@@ -438,12 +609,24 @@ def _random_model_name():
 @dataclasses.dataclass(frozen=True)
 class Model:
     name: str = dataclasses.field(default_factory=_random_model_name)
+
     uuid: str = dataclasses.field(default_factory=lambda: str(uuid4()))
 
     # whatever juju models --format=json | jq '.models[<current-model-index>].type' gives back.
     # TODO: make this exhaustive.
     type: Literal["kubernetes", "lxd"] = "kubernetes"
 
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        *,
+        uuid: Optional[str] = None,
+        type: Literal["kubernetes", "lxd"] = "kubernetes",
+    ):
+        object.__setattr__(self, "name", _random_model_name() if name is None else name)
```

Comment:
```suggestion
        object.__setattr__(self, "name", name or _random_model_name())
```
I'm quite keen on this syntax, and you seem to also use it in other places. should we be consistent?

Reply:
I did this to continue to allow a name of `""`, although I assume that can never be the case in Juju and it probably therefore doesn't make sense here.

I do prefer the latter as well. This code is vanishing anyway, though.